libminifi/test/unit/SiteToSiteTests.cpp

```diff
@@ -0,0 +1,343 @@
+
+  auto fake_uuid = minifi::utils::Identifier::parse("C56A4180-65AA-42EC-A945-5FD21DEC0538").value();
+  protocol.setPortId(fake_uuid);
+
+  verifyBootstrapMessages(protocol, *collector_ptr);
+
+  // start to send the stuff
+  auto transaction = SiteToSiteClientTestAccessor::createTransaction(protocol, sitetosite::TransferDirection::SEND);
+  REQUIRE(transaction);
+  collector_ptr->get_next_client_response();
+  REQUIRE(collector_ptr->get_next_client_response() == "SEND_FLOWFILES");
+
+  TestController test_controller_;
+  TestController::PlanConfig plan_config_;
+  std::shared_ptr<TestPlan> test_plan = test_controller_.createPlan(plan_config_);
+  test_plan->addProcessor("DummyProcessor", "dummyProcessor");
+  std::shared_ptr<minifi::core::ProcessContext> context = [&] { test_plan->runNextProcessor(); return test_plan->getCurrentContext(); }();
+  std::unique_ptr<minifi::core::ProcessSession> session = std::make_unique<core::ProcessSessionImpl>(context);
+
+  auto flow_file = session->create();
+  session->write(flow_file, [&](const std::shared_ptr<io::OutputStream>& output_stream) {
+    std::span<const std::byte> span{reinterpret_cast<const std::byte*>(payload.data()), payload.size()};
+    output_stream->write(span);
+    return payload.size();
+  });
+  flow_file->updateAttribute("filename", "myfile");
+  flow_file->updateAttribute("flow.id", "test");
+  session->transfer(flow_file, DummyProcessor::Success);
+  session->commit();
+
+  std::map<std::string, std::string> attributes;
+  REQUIRE(SiteToSiteClientTestAccessor::sendFlowFile(protocol, transaction, *flow_file, *session));
+  verifySendResponses(*collector_ptr, expected_responses);
+  REQUIRE(transaction->getCRC() == 2886786428);
+}
+
+TEST_CASE("TestSiteToSiteVerifyNegotiationFail", "[S2S]") {
+  auto collector = std::make_unique<SiteToSiteResponder>();
+
+  const char negotiated_abort_code = magic_enum::enum_underlying(sitetosite::ResourceNegotiationStatusCode::NEGOTIATED_ABORT);
+  std::string resp_code;
+  resp_code.insert(resp_code.begin(), negotiated_abort_code);
+  collector->push_response(resp_code);
+  collector->push_response(resp_code);
+
+  auto peer = std::make_unique<sitetosite::SiteToSitePeer>(std::move(collector), "fake_host", 65433, "");
+  sitetosite::RawSiteToSiteClient protocol(std::move(peer));
+
+  std::string uuid_str = "C56A4180-65AA-42EC-A945-5FD21DEC0538";
+  minifi::utils::Identifier fake_uuid;
+  fake_uuid = uuid_str;
+  protocol.setPortId(fake_uuid);
+
+  REQUIRE_FALSE(SiteToSiteClientTestAccessor::bootstrap(protocol));
+}
+
+void iniitalizeMockRemoteClientReceiveDataResponses(SiteToSiteResponder& collector) {
+  collector.push_response("R");
+  collector.push_response("C");
+  auto addResponseCode = [&collector](sitetosite::ResponseCode code) {
+    std::string resp_code;
+    resp_code.insert(resp_code.begin(), magic_enum::enum_underlying(code));
+    collector.push_response(resp_code);
+  };
+  addResponseCode(sitetosite::ResponseCode::MORE_DATA);
```

Comment:
I would rewrite this to
```suggestion
    collector.push_response(std::string{static_cast<char>(magic_enum::enum_underlying(sitetosite::ResponseCode::MORE_DATA))});
```
because why use 6 lines of code where 1 will do